sonar-plugin-api/src/main/java/org/sonar/api/resources/File.java

```diff
@@ -19,22 +19,21 @@
  */
 package org.sonar.api.resources;
 
+import java.util.List;
+import javax.annotation.CheckForNull;
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.builder.ToStringBuilder;
 import org.sonar.api.batch.SensorContext;
 import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.scan.filesystem.PathResolver;
 import org.sonar.api.utils.WildcardPattern;
 
-import javax.annotation.CheckForNull;
-
-import java.util.List;
-
 /**
- * This class is an implementation of a resource of type FILE
- *
  * @since 1.10
+ * @deprecated since 5.6 replaced by {@link InputFile}.
  */
+@Deprecated
 public class File extends Resource {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)